ntbs-service/Services/LegacySearchBuilder.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Dynamic;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+using ntbs_service.Models;
+using ntbs_service.Models.Entities;
+
+namespace ntbs_service.Services
+{
+    public interface ILegacySearchBuilder : ISearchBuilder
+    {
+        (string, dynamic) GetResult();
+    }
+
+    public class LegacySearchBuilder : ILegacySearchBuilder
+    {
+        string sqlQuery;
+        dynamic parameters;

```

Comment:
Using a dynamic to add the different parameters onto the object as the builder is formed - I could abstract this to an interface to avoid using dynamic but without any other benefit I can see right now. I'm not sure how the sql query in LegacySearchService would work with a specific class instead of a general object (hopefully sensibly).

Reply:
It *seems* right to me  - especially as we're not reading the object directly ourselves after that!